### PR TITLE
Added authentication

### PR DESCRIPTION
**Pull request by bitsandfoxes**

### CODE REVIEW

File: Sentaur.Leaderboard.Api/Program.cs [hunk 1]
```diff
@@ -64,12 +117,34 @@
...
+app.MapPost("/getToken", [AllowAnonymous](User user) =>
+{
+    if (user is not { UserName: "user1", Password: "password1" })
+    {
+        return Results.Unauthorized();
+    }
+    
+    var issuer = builder.Configuration["Jwt:Issuer"];
+    var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Jwt:Key"] ?? throw new InvalidOperationException("Failed to get 'Jwt:Key'")));
+    var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
```
Comment (bruno-garcia) [inline Sentaur.Leaderboard.Api/Program.cs]: could this be done outside this closure? We'll be using this on each getToken request and it'll be ready to go

File: Sentaur.Leaderboard.Api/Program.cs [hunk 2]
```diff
@@ -64,12 +117,34 @@
...
+
+    var token = new JwtSecurityToken(issuer: issuer, signingCredentials: credentials);
+    var tokenHandler = new JwtSecurityTokenHandler();
+        
+    var stringToken = tokenHandler.WriteToken(token);
+
+    return Results.Ok(stringToken);
+});
+
+app.UseAuthentication();
```
Comment (bruno-garcia) [inline Sentaur.Leaderboard.Api/Program.cs]: Should these be before the endpoints?

File: Sentaur.Leaderboard.Api/User.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace Sentaur.Leaderboard.Api;
+
+public class User
+{
+    public string UserName { get; set; }
```
Comment (bruno-garcia) [inline Sentaur.Leaderboard.Api/User.cs]: username is one word

```suggestion
    public string Username { get; set; }
```

File: Sentaur.Leaderboard.Api/Program.cs (hunk 1 above)
Comment (bitsandfoxes) [reply Sentaur.Leaderboard.Api/Program.cs]: Done

File: Sentaur.Leaderboard.Api/Program.cs (hunk 2 above)
Comment (bitsandfoxes) [reply Sentaur.Leaderboard.Api/Program.cs]: As far as I could figure out: It is only sensitive to `UseEndpoints` and `UseRouting` and seems to work before **and** after the `app.MapX` calls, which I don't quite get.

**Pull request merged**